src/topology/algebra/order/lower.lean

```diff
@@ -38,19 +38,18 @@ open set topological_space
 /--
 Type synonym for a preorder equipped with the lower topology
 -/
-@[reducible]
-def lower (α : Type u) := α
+@[derive preorder, nolint unused_arguments]
+def lower (α : Type u) [preorder α] := α
 
--- Needed if the above definition is not reducible
-/-
-instance lower.preorder (α : Type u) [preorder α] :
-  preorder (lower α) := _inst_1
--/
+instance (α : Type u) [preorder α] [p : nonempty α] : nonempty (lower α) := p
+
+instance (α : Type u) [preorder α] [p : inhabited α] : inhabited (lower α) := p
 
 /--
 The lower topology is the topology generated by the complements of the closed intervals to infinity
 -/
-def lower_topology (α : Type u) [preorder α] : topological_space α := generate_from {s | ∃ a, (Ici a)ᶜ = s}
+def lower_topology (α : Type u) [preorder α] : topological_space α :=
+  generate_from {s | ∃ a, (Ici a)ᶜ = s}
```

Comment:
```suggestion
generate_from {s | ∃ a, (Ici a)ᶜ = s}
```

Reply:
@alreadydone  Should this not be indented according to the [style guide](https://leanprover-community.github.io/contribute/style.html#structuring-definitions-and-theorems)?

Reply:
From the style guide:

> After stating the theorem, we generally do not indent the first line of a proof, so that the proof is "flush left" in the file.

> for "proof" also read "definition body".